collection-scripts/gather_monitoring

```diff
@@ -1,29 +1,235 @@
+fi
+
+_START=${GAHTER_MONIT_START_DATE:-${GAHTER_MONIT_START_DATE_DEFAULT}}
+DATE_START=$(date -d "${_START}" +%s)
+DATE_END=$(date +%s)
+# If the resolution is too low, the API will deny due to 11k limitation
+QUERY_STEP=${GAHTER_MONIT_QUERY_STEP:-${GAHTER_MONIT_QUERY_STEP_DEFAULT}}
+
+echo "INFO: Metrics config: from=${DATE_START} to=${DATE_END} step=${QUERY_STEP}"
+echo "INFO: Config Env GAHTER_MONIT_START_DATE: ${GAHTER_MONIT_START_DATE}"
+echo "INFO: Config Env GAHTER_MONIT_QUERY_STEP: ${GAHTER_MONIT_QUERY_STEP}"
+echo "INFO: Config Env GATHER_MONIT_CUSTOM_METRICS: ${GATHER_MONIT_CUSTOM_METRICS}"
+echo "INFO: Config Env GATHER_MONIT_DISCOVERY_PREFIXES: ${GATHER_MONIT_DISCOVERY_PREFIXES}"
+echo "INFO: Config Env GATHER_MONIT_DISCOVERY_GRAFANA: ${GATHER_MONIT_DISCOVERY_GRAFANA}"
+echo "INFO: Prometheus endpoint URL: ${PROM_URL}"
+echo "INFO: Grafana  endpoint URL: ${GF_URL}"
+
+function install_jq() {
+
+    if [[ -f ${JQ_PATH} ]]; then
+        return
+    fi
+
+    # install jq from binary (required to Dashnoard parser)
+    export JQ_PATH=/usr/local/bin/jq
+    curl -s -o ${JQ_PATH} \
+      http://stedolan.github.io/jq/download/linux64/jq && \
+      chmod +x ${JQ_PATH}
+}
+
+# Make a query (single metric) to Prometheus API /api/v1/query_range
+function prom_query_range() {
+    QUERY=$1
+    METRIC_FILE=${2:-$(echo ${QUERY} |awk -F'\(' '{print$1}')}
+
+    # The metrics are ziped to decrease the datatransfer
+    curl ${CURL_OPTIONS} \
+        -H "Authorization: Bearer ${OCP_TOKEN}" \
+        -H 'Accept-encoding: gzip' \
+        --data-urlencode "query=${QUERY}" \
+        --data-urlencode "start=${DATE_START}" \
+        --data-urlencode "end=${DATE_END}" \
+        --data-urlencode "step=${QUERY_STEP}" \
+        $PROM_URL/api/v1/query_range > ${PROM_PATH}/metric-${METRIC_FILE}.json.gz
+}
+
+# Gather metrics defined on environment GATHER_MONIT_CUSTOM_METRICS
+function get_custom_metrics() {
+
+    if [[ -z "${GATHER_MONIT_CUSTOM_METRICS}" ]]; then
+        return
+    fi
+
+    echo "INFO: Metrics will be collected from [$(date -d "@${DATE_START}")] to [$(date -d "@${DATE_END}")]"
+    for METRIC in $GATHER_MONIT_CUSTOM_METRICS; do
+        echo "INFO: Getting metric range: $METRIC"
+        prom_query_range "$METRIC" "$METRIC"
+    done
+}
+
+# Discovery metrics from prefixes defined on environment GATHER_MONIT_DISCOVERY_PREFIXES
+function get_discovery_metrics() {
+
+    if [[ -z "${GATHER_MONIT_DISCOVERY_PREFIXES}" ]]; then
+        return
+    fi
+
+    echo "INFO: Starting discovery metrics by prefixes..."
+    # TODO: fix it shame~~ O(n*m) ;(
+    while read METRIC; do
+        for PREFIX in ${GATHER_MONIT_DISCOVERY_PREFIXES}; do
+            if [[ "${METRIC}" =~ ^${PREFIX}.* ]]; then
+                echo "INFO: Getting metric range: ${METRIC}"
+                prom_query_range "${METRIC}" "${METRIC}" ;
+            fi
+        done
+    done < <( curl -sk ${CURL_OPTS} \
```

Comment:
Double quotes around CURL_OPTS.

Reply:
I changed to the curl-ing global var. 👍🏼